src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1233,7 +1261,7 @@ private BoundExpression BindDefaultExpression(DefaultExpressionSyntax node, Diag
             TypeWithAnnotations typeWithAnnotations = this.BindType(node.Type, diagnostics, out AliasSymbol alias);
             var typeExpression = new BoundTypeExpression(node.Type, aliasOpt: alias, typeWithAnnotations);
             TypeSymbol type = typeWithAnnotations.Type;
-            return new BoundDefaultExpression(node, typeExpression, constantValueOpt: type.GetDefaultValue(), type);
+            return new BoundConvertedDefaultExpression(node, naturalTypeOpt: type, typeExpression, constantValueOpt: type.GetDefaultValue(), type);
```

Comment:
>BoundConvertedDefaultExpression [](start = 23, length = 31)

This feels unexpected. I would expect only DefaultLiteralExpression require "Converted" form.